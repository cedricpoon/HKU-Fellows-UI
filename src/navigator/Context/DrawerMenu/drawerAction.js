import { link, profile } from 'hkufui/config/webapi';
import { LOGOUT, UPDATE_TEMPERATURE } from 'hkufui/src/constants/actionTypes';

export function onLogout() {
  return (dispatch) => {
    dispatch({ type: LOGOUT });
  };
}

export function onLoadUserTemperature() {
  return async (dispatch, getState) => {
    const { credential } = getState();
    try {
      const response = await fetch(link(profile.temperature()), {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          username: credential.userId,
          token: credential.token
        }),
      });
      const res = await response.json();
      if (res.status === 200) {
        // success
        dispatch({
          type: UPDATE_TEMPERATURE,
          payload: { ...res.payload }
        });
      }
    } catch (error) { /* ignored */ }
  };
}

const handleActions = (state = {}, action = {}) => {
  switch (action.type) {
    case UPDATE_TEMPERATURE:
      return {
        ...state,
        temperature: action.payload.temperature
      }
    default:
      return state;
  }
};

export default handleActions;
