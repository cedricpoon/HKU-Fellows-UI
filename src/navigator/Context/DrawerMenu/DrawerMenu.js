import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Drawer } from 'hkufui/components';
import { onLogout } from './drawerAction';

export class DrawerMenu extends PureComponent {
  render() {
    // restProps as required in DrawerItems
    const { onLogout, ...restProps } = this.props;
    return (
      <Drawer onLogout={onLogout} {...restProps} />
    );
  }
}

DrawerMenu.propTypes = {
  onLogout: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => ({
  onLogout: () => { dispatch(onLogout()) }
})

export default connect(
  null,
  mapDispatchToProps
)(DrawerMenu);
