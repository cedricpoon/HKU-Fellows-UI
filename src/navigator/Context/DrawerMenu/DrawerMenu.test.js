import React from 'react';
import { shallow } from 'enzyme';
import DrawerMenu from './DrawerMenu';

describe('Testing DrawerMenu Component', () => {
  const requiredProps = {
    onLogout: () => {}
  };

  it('renders as expected', () => {
    const wrapper = shallow(
      <DrawerMenu {...requiredProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
