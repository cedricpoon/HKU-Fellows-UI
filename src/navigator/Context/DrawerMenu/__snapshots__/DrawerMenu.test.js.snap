// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DrawerMenu Component renders as expected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
